src/index/vocabulary/VocabularyInMemoryBinSearch.h

```diff
@@ -71,15 +71,19 @@ class VocabularyInMemoryBinSearch
         uint64_t, ad_utility::serialization::FileWriteSerializer>;
     OffsetWriter offsetWriter_;
     std::optional<uint64_t> lastIndex_ = std::nullopt;
+    std::string readableName_ = "";
     // Construct a `WordWriter` that will write to the given `filename`.
     explicit WordWriter(const std::string& filename);
     // Add the given `word` with the given `idx`. The `idx` must be greater than
     // all previous indices.
-    void operator()(std::string_view word, uint64_t idx);
+    uint64_t operator()(std::string_view word, bool isExternalDummy = false,
+                        uint64_t idx = 0);
```

Comment:
That doesn't make sense and is dangerous.
This vocabulary is different and can't be used like the others,
so please leave its interface as it was. In particular we can't have a default value for the `idx`.